File: android_compose/app/src/main/java/com/rencloud/app/data/RenCloudPlan.kt
package com.rencloud.app.data

enum class BillingCycle { MONTHLY, ANNUAL }

enum class AppCurrency(val symbol: String, val rate: Double) {
    INR("₹", 1.0),
    USD("$", 0.012),
    EUR("€", 0.011),
    AED("Dh ", 0.044);

    fun format(priceInr: Int): String {
        return when (this) {
            INR -> "$symbol$priceInr"
            USD -> "$symbol${String.format("%.2f", priceInr * rate)}"
            EUR -> "$symbol${String.format("%.2f", priceInr * rate)}"
            AED -> "$symbol${String.format("%.1f", priceInr * rate)}"
        }
    }
}

data class RenCloudPlan(
    val id: String,
    val name: String,
    val categoryId: String,
    val categoryName: String,
    val ram: String,
    val nvmeStorage: String,
    val cpu: String,
    val monthlyPriceInr: Int,
    val isPopular: Boolean = false,
    val tierType: String? = null,
    val databases: Int? = null,
    val backups: Int? = null,
    val isOneTime: Boolean = false,
    val extraInfo: String? = null
) {
    fun getPriceForCycle(cycle: BillingCycle): Int {
        if (isOneTime) return monthlyPriceInr
        return if (cycle == BillingCycle.ANNUAL) {
            (monthlyPriceInr * 0.85).toInt()
        } else {
            monthlyPriceInr
        }
    }
}

File: lib/models/rencloud_plan.dart
enum BillingCycle { monthly, annual }

class RenCloudPlan {
  final String id;
  final String name;
  final String categoryId;
  final String categoryName;
  final String ram;
  final String nvmeStorage;
  final String cpu;
  final int monthlyPriceInr;
  final bool isPopular;
  final String? tierType; // Budget, Premium, Enterprise
  final int? databases;
  final int? backups;
  final String? extraInfo;
  final bool isOneTime; // For setup services

  const RenCloudPlan({
    required this.id,
    required this.name,
    required this.categoryId,
    required this.categoryName,
    required this.ram,
    required this.nvmeStorage,
    required this.cpu,
    required this.monthlyPriceInr,
    this.isPopular = false,
    this.tierType,
    this.databases,
    this.backups,
    this.extraInfo,
    this.isOneTime = false,
  });

  int getPriceForCycle(BillingCycle cycle) {
    if (isOneTime) return monthlyPriceInr;
    if (cycle == BillingCycle.annual) {
      // 15% discount for annual billing
      return (monthlyPriceInr * 0.85).round();
    }
    return monthlyPriceInr;
  }
}
